internal/operator/controller.go

```diff
@@ -353,6 +428,28 @@ func (c *Controller) createBootstrapPod(o *stanv1alpha1.NatsStreamingCluster) er
 	return nil
 }
 
+func (c *Controller) createPodFrom(o *stanv1alpha1.NatsStreamingCluster, pod *k8scorev1.Pod) (*k8scorev1.Pod, error) {
```

Comment:
@wallyqs There is some code duplication between this method and `createMissingPods`. I was not immediatedly sure how to attack that given the different intentions of each.

Reply:
Thanks a lot for the PR and the updates, I will take a look soon. We can bring this changes in the next release, but what I was thinking that another way to fix the reconcile issues is to make the operator instead of creating each one of the pods, it creates a statefulset that represents those pods.  That would also help make it possible to use the per pod volume claim template feature which is only available for statefulsets.

Reply:
Yes, passing the changes from the template to a statefulset would be cleaner saving on the direct pod creation/delete logic. Can be a future improvement. I was just focused on making my use case work within the current way things were working.